ls1intum/Artemis
# `Communication`: Hide notifications for active channel on lecture and exercise page

### Checklist
#### General


#### Client

### Motivation and Context


Notifications are shown for the active channel on the lecture/exercise detail pages, even though the channel is open.

### Description

Notifications for the open channel are temporarily disabled as long as the sidebar is visible

### Steps for Testing

Prerequisites:
- 2 Students
- 1 Course with messaging enabled

1. Go to the same exercise/lecture details page with both students
2. Write messages as student 1
3. Student 2 should see the message, but no notification
4. Go to the messages tab with student 1 and write a message in a different channel (which student 2 is also a member of).
5. Student 2 should see a notification for this

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [discussion-section.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5617/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/discussion-section/discussion-section.component.ts.html) | 86.77% | ✅ |
| [notification.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5617/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/notification/notification.service.ts.html) | 75% | ✅ |
| [notification-settings.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5617/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/user-settings/notification-settings/notification-settings.service.ts.html) | 94.44% | ✅ |

### Screenshots


No UI changes
